lib/membrane_rtc_engine/engine.ex

```diff
@@ -832,6 +753,21 @@ defmodule Membrane.RTC.Engine do
         )
 
       state = put_in(state, [:peers, peer_id], peer)
+
+      peers =
+        Map.new(state.peers, fn {peer_id, peer} ->
+          peer = put_track_id_to_track_metadata_in_peer(peer, state.endpoints)
+          {peer_id, peer}
+        end)
+
+      MediaEvent.create_peer_accepted_event(peer_id, Map.delete(peers, peer_id))
+      |> dispatch()
+
+      peer = put_track_id_to_track_metadata_in_peer(peer, state.endpoints)
+
+      MediaEvent.create_peer_joined_event(peer_id, peer)
+      |> dispatch()
```

Comment:
Actually why do we still use `track_id_to_track_metadata`? Can't we store this in `Track.t()`? We can also use get these metadata from `Endpoint.t()` we don't have to put this to peer to create `peer_joined_event`

Reply:
This field is not stored in Peer struct. Metadata is stored inside `Track.t()`. We put this field inside peer only for these events, because it is better to put one field inside Peer before sending event than creating map 
```elixir
%{
id: peer.id,
metadata: peer.id,
track_id_to_track_metadata: Endpoint.get_track_id_to_track_metadata(...)
}
```

Reply:
```elixir
  defp do_accept_new_peer(peer_id, data, state) do
    if Map.has_key?(state.peers, peer_id) do
      Membrane.Logger.warn("Peer with id: #{inspect(peer_id)} has already been added")
      {[], state}
    else
      peer =
        Peer.new(
          peer_id,
          data.metadata
        )

      state = put_in(state, [:peers, peer_id], peer)

      MediaEvent.create_peer_accepted_event(peer_id, Map.delete(peers, peer_id), endpoints)
      |> dispatch()

      MediaEvent.create_peer_joined_event(peer_id, peer)
      |> dispatch()

      {[], state}
    end
  end

  def create_peer_accepted_event(peer_id, peers, endpoints) do
    peers =
      Enum.map(peers, fn {id, peer} ->
        track_id_to_track_metadata = Endpoint.get_track_id_to_metadata(endpoints[id])
        %{id: id, metadata: peer.metadata, trackIdToMetadata: track_id_to_track_metadata}
      end)

    %{
      type: "peerAccepted",
      data: %{
        id: peer_id,
        peersInRoom: peers
      }
    }
    |> do_create(peer_id)
  end
  
  def create_peer_joined_event(peer_id, peer) do
    %{
      type: "peerJoined",
      data: %{
        peer: %{
          id: peer_id,
          metadata: peer.metadata,
        }
      }
    }
    |> do_create(:broadcast)
  end
 ``` 
This should work. There shouldn't be track_id_to_metadata in peer_joined